### ISSUE TITLE
Apple Music using #AlbumView causes crashes

### ISSUE THREAD

**Issue by tagdara** [SoCo/SoCo#399]
Apple Music has a number of issues with Sonos, and this one seems to also affect the Windows Desktop Controller. 

When you Queue up an entire album, Soco fails on the first track change with an unknown UPNP type of object.container.album.musicAlbum.#AlbumView

Since I convert everything down to regular dicts in my app, it's very frustrating every time a new service is added which does something new.  Incidentally, the use of #AlbumView is even breaking the Windows controller, causing the player to stop and the queue to jump to the first track and loop the first 1-2 seconds over and over.  So this is clearly not just a Soco issue but rather a larger implementation issue with Sonos and Didl.

I'm assuming this has the same attributes as musicAlbum, so I monkeypatched data_structures with this which seems to resolve it for me, but I'm not sure if there are better ways to handle the continously changing Didl datastructures.  

class DidlMusicAlbumView(DidlAlbum):

    """Class that represents a music library album. """

    item_class = 'object.container.album.musicAlbum.#AlbumView'
    # According to the spec, all musicAlbums should be represented in
    # XML by a <container> tag. Sonos sometimes uses <container> and
    # sometimes uses <item>. Set the tag type to '' to indicate that
    # either is allowed.
    tag = ''
    # name: (ns, tag)
    # pylint: disable=protected-access
    _translation = DidlAudioItem._translation.copy()
    _translation.update(
        {
            'artist': ('upnp', 'artist'),
            'genre': ('upnp', 'genre'),
            'producer': ('upnp', 'producer'),
            'toc': ('upnp', 'toc'),
            'album_art_uri': ('upnp', 'albumArtURI'),
        }
    )

**Comment by tagdara** [comment]
After looking at this deeper and reading about the Deezer issues in #388, it looks like the use of .#subclass is a thing, whether it's official or not.  While I originally hit this with #AlbumView, now I've seen a few others like #SongTitleWithArtist.  All of these seem to work fine when they're pointed to their parent class.  

So in order to hopefully fix this in a more extensive fashion I made the following changes to data_structures in from_didl_string:

```
            try:
                if item_class in _DIDL_CLASS_TO_CLASS:
                        cls = _DIDL_CLASS_TO_CLASS[item_class]
                elif item_class.find(".#")>-1:
                        print  item_class[:item_class.find(".#")]
                        if item_class[:item_class.find(".#")] in _DIDL_CLASS_TO_CLASS:
                                cls = _DIDL_CLASS_TO_CLASS[item_class[:item_class.find(".#")]]
            except KeyError:
                raise DIDLMetadataError("Unknown UPnP class: %s" % item_class)

``` 

I then also replaced line 371 to prevent the mismatched class detection from killing my fix:
```
if item_class.find(cls.item_class)!=0:
#if item_class != cls.item_class:
```

So far so good.  I've been randomly grabbing songs from the Apple Music catalog without any crashes this morning, so at least for my scenario it's working better.

**Comment by KennethNielsen** [comment]
Hallo @tagdara 

Finally got around to loking at this. I like your idea about simply assuming that the parent class will do and discard everything after. Do you want to turn your fix into a PR? Otherwise I can do it for you.

Regards Kenneth

**Comment by KennethNielsen** [comment]
Hallo @tagdara 

I will start working on a PR based on your solution.

Regards Kenneth

**Comment by KennethNielsen** [comment]
Hallo @tagdara 

I have tried to implement your fix in the branch (https://github.com/KennethNielsen/SoCo/tree/pound_subclass_fix) requested for pull in #425.

Can you test it?

Regards Kenneth

**Comment by KennethNielsen** [comment]
This should now hopefully be fixed with #425. Thanks for reporting it and for the idea for the fix.
